### PR TITLE
xcodegen post

### PR DESCRIPTION
**Pull request by Lutzifer**

### CODE REVIEW

**Comment by Lutzifer** [comment]
@hmSchuller Could you please follow along the tutorial to make sure everything works and make screenshots on the way to illustrate the article?

File: source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    type: application
+    platform: iOS
+    sources:
+      #Sources
+      - path: XcodegenApp
+    settings:
+      IPHONEOS_DEPLOYMENT_TARGET: 10.3
+      PRODUCT_BUNDLE_IDENTIFIER: "de.number42.XcodegenApp"
+      PRODUCT_NAME: "XcodegenApp"
+      INFOPLIST_FILE: "XcodegenApp/Info.plist"
```
Comment (yonaskolb) [inline source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md]: `Info.plist` is automatically found

File: source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+  XcodegenApp:
+    type: application
+    platform: iOS
+    sources:
+      #Sources
+      - path: XcodegenApp
+    settings:
+      IPHONEOS_DEPLOYMENT_TARGET: 10.3
+      PRODUCT_BUNDLE_IDENTIFIER: "de.number42.XcodegenApp"
+      PRODUCT_NAME: "XcodegenApp"
```
Comment (yonaskolb) [inline source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md]: Product name is automatically set

File: source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+targets: # The List of our targets
+  XcodegenApp:
+    type: application
+    platform: iOS
+    sources:
+      #Sources
+      - path: XcodegenApp
+    settings:
+      IPHONEOS_DEPLOYMENT_TARGET: 10.3
+      PRODUCT_BUNDLE_IDENTIFIER: "de.number42.XcodegenApp"
```
Comment (yonaskolb) [inline source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md]: You could use `options.bundleIdPrefix` of `de.number42`

File: source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+  tabWidth: 2 # a tab is 2 spaces
+targets: # The List of our targets
+  XcodegenApp:
+    type: application
+    platform: iOS
+    sources:
+      #Sources
+      - path: XcodegenApp
+    settings:
+      IPHONEOS_DEPLOYMENT_TARGET: 10.3
```
Comment (yonaskolb) [inline source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md]: You can use `deploymentTarget: "10.3"` on the target

File: source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+Generate the project, open it and run the tests: voila!
+
+To add UI Tests, add this target:
+
+```yml
+XcodegenApp-iOS-UITests:
+  type: bundle.ui-testing
+  platform: iOS
+  settings:
+    BUNDLE_LOADER: "$(TEST_HOST)"
```
Comment (yonaskolb) [inline source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md]: The `BUNDLE_LOADER: "$(TEST_HOST)"` is automatically set through setting presets

File: source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+## Generating TestTargets
+
+Add the following target to the project.yml:
+
+```yml
+XcodegenApp-iOS-Tests:
+  type: bundle.unit-test
+  platform: iOS
+  settings:
+    TEST_HOST: "$(BUILT_PRODUCTS_DIR)/XcodegenApp.app/XcodegenApp"
```
Comment (yonaskolb) [inline source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md]: I don't think this `TEST_HOST` is required

Comment (Lutzifer) [reply source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md]: Had to do this in the whitelabeling project we have, it is indeed not required here.

File: source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+
+- Though you can manually resolve merge conflicts in this file, e.g. if source code files or resources have been added on different branches, you can never be sure if the file is correct afterwards.
+- Syncing the folder structure on disk and the group structure in the project is mainly a manual process that sometimes leads to confusion. There already are tools to mitigate this, like `synx` or the sort functionality of the `xcodeproj` gem.
+- Xcode does not warn you when a file is missing until you compile.
+- Orchestrating dependencies and build scripts of multiple targets can become quite a hassle.
+
+## Introducing Xcodegen
+
+Xcodegen is a tool, that allows us to generate the xcodeproj file from a definition in a file called project.yml. As the xcodeproj file can be generated whenever we like, we do not even have to keep it inside our git and can ignore it (though I personally prefer to keep it checked in, so I can see what changes my edits to the project.yml file introduced to the project).
```
Comment (yonaskolb) [inline source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md]: Maybe this first reference to XcodeGen can be a link? 😄

Comment (Lutzifer) [reply source/blog/articles/1970-01-01-xcodegen/1970-01-01-xcodegen.html.md]: I think it was one, once upon a time, before converting the text to markdown

**Pull request merged**